### DISCUSSION TITLE
Is rtcGetLocalDescription *always* available straight away after rtcSetLocalDescription?

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#1116] [General]
Is there any case when a significant delay between calling `rtcSetLocalDescription()` and `rtcGetLocalDescription()` being available occurs? From experimenting with it seems it's always instant, but just wondering if there's any scenario where it may not be?
Why I'm asking: I have a GUI program from which I'm using libdatachannel. Fewer callbacks can simplify my code significantly (if it always returns instantaneously I could just block the main thread to wait)

**Comment by paullouisageneau** [comment]
There is no guarantee. In practice, STUN and TURN servers won't have answered yet so candidates gathering won't be finished, and the connection will typically fail if not connecting locally. You must either wait for the gathering state to be complete or get the local description then local candidates from their respective callbacks. Don't block the main thread to wait as most things involve network operations.
